### DISCUSSION TITLE
可以本版块提出软件的bug或者想要的功能

### DISCUSSION THREAD

**Discussion by ywmoyue** [ywmoyue/biliuwp-lite#13] [Announcements]
如果是代码上的问题，请提交issue
本贴是公告，尽量不要在本贴提bug

**Comment by mikazukizero** [comment]
下载视频导出功能有bug，导出结果是一个0B的文件。

**Reply by ywmoyue** [reply]
我打包方式有问题，下个版本恢复

**Comment by konata422** [comment]
发现页点不开，或许是跨域了？

**Reply by ywmoyue** [reply]
发现页的数据是这玩意儿： https://github.com/ywmoyue/biliuwp-lite/blob/master/document/entrance.json
点不开可能是网络访问不了github

**Comment by PRO-2684** [comment]
- 希望鼠标中键的操控方式向浏览器进一步靠近，具体可以改进以下两点
    - 首页视频/视频推荐中键点击后在新标签页打开，但是仍然停留在当前标签页
    - 禁用页面任意位置中键均会关闭当前标签页的操作
- 首页视频强制显示 up 主 (目前版本有一部分视频会显示 tag 而不是 up 主)

**Reply by ywmoyue** [reply]
第一点在做了，预计下个版本出，第二点还没看

**Comment by suorumei** [comment]
请问可以增加合集功能吗

**Reply by suorumei** [reply]
还有可以增加自选弹幕字体功能就好了

**Reply by ywmoyue** [reply]
合集功能是有的，在“我的收藏”里面有“我的收藏与订阅”就是订阅的合集了，后续我看看优化一下吧

**Comment by Jiagu1218** [comment]
看能不能添加个弹幕字体调整的功能，参考云之幻版本的

**Comment by Haibaraxxx** [comment]
搜索功能有望修复吗

**Reply by ywmoyue** [reply]
额，搜索功能有啥问题吗，我自己用的是好的

**Reply by Haibaraxxx** [reply]
> 额，搜索功能有啥问题吗，我自己用的是好的

![QMO}(XXYAQ5IWC0(__%I VN](https://user-images.githubusercontent.com/121796816/219330548-6f08a827-26f3-4927-b329-7615d2081715.png)

**Reply by ywmoyue** [reply]
感谢反馈，我测出来了，不登陆使用不了搜索，我一直是登录着用的没发现

**Reply by Haibaraxxx** [reply]
> 感谢反馈，我测出来了，不登陆使用不了搜索

我已经登陆了。逍遥原版也有这样的问题。当时是输入B站网址https://www.bilibili.com/可以解决，只不多 一段是时间后又不行了

**Reply by Haibaraxxx** [reply]
> 感谢反馈，我测出来了，不登陆使用不了搜索，我一直是登录着用的没发现

貌似是扫码登陆会出现，我刚才用密码登陆已经解决了。谢谢作者

**Comment by Haibaraxxx** [comment]
可以加入高级弹幕吗

**Reply by Leinbow** [reply]
@ywmoyue 你好，能否换成暗影吉他手写的开源弹幕引擎吗？它能支持高级弹幕。
> https://github.com/cotaku/DanmakuFrostMaster

**Comment by mikazukizero** [comment]
下载视频导出有Bug，flv和H.265都有，其余项目未测试。报错为：

执行任务失败：
未能加载文件或程序集"System.Runtime.CompilerServices.Unsafe, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"或者它的某一个依赖项。系统找不到指定文件。

使用NuGet安装system.runtime.compilerservices.unsafe后依旧报错。
没有尝试过重启电脑。

**Reply by ywmoyue** [reply]
感谢反馈，已记录

**Reply by ywmoyue** [reply]
依赖路径错误的问题，dev分支已修复，暂时不发版了，可以自行clone编译

**Comment by FadeJay** [comment]
有的视频评论加载失败，比如BV18Y411e7bt

**Reply by ywmoyue** [reply]
https://github.com/ywmoyue/biliuwp-lite/discussions/20

**Reply by ywmoyue** [reply]
4.5.2已修复

**Reply by FadeJay** [reply]
谢谢作者

**Comment by enterlet** [comment]
能不能有个记住铺满窗口状态的功能，大致描述下就是点视频，进入默认就是铺满窗口状态，取消全屏，也是进入铺满窗口状态

**Comment by Haibaraxxx** [comment]
![B0J}5$8UD)B5Y%QCQVEZOHT](https://user-images.githubusercontent.com/121796816/221388493-b38d70e8-9f60-49ac-b428-30dcc6c11aba.png)
清晰度最高好像只有480p唉

**Reply by FadeJay** [reply]
你没有登录账号吧

**Reply by ywmoyue** [reply]
偶现登录账号也只有480p，我之后没复现出来了，重新登录一下就可以了

**Comment by ybhgl** [comment]
https://www.bilibili.com/video/BV1tM411E74g/ 这个视频点进去显示 啥都木有（具体点说这个up主的所有视频都打不开）
还有就是这个up主我已经关注了，但是在软件里的关注列表找不到这个人，网页端正常

**Reply by ywmoyue** [reply]
这就api限制了，用原来的api拿不到这个视频的数据，我给换个api吧，关注列表暂时没想到啥好办法

**Reply by ywmoyue** [reply]
4.5.3修复该视频打不开问题

**Comment by liuyang19990813** [comment]
希望适配长按右键倍速播放（类似网页版那样子）。非常实用的功能呢。

**Comment by Ech023** [comment]
4.6.2版本，就是用户登录后用ccleaner清楚垃圾文件数据后，uwp用户登录还在，但是视频的可选择的视频清晰度只有480p和360p了。没有正常登录时1080极其更高清楚度。

**Reply by Ech023** [reply]
需要重新登陆，然后再登陆才能到正常清晰度。

**Reply by ywmoyue** [reply]
4.6.3之后是否仍会出现这个问题？

**Reply by Ech023** [reply]
> 4.6.3之后是否仍会出现这个问题？

现在还是存在这种问题，应该是cookie被清除了。

**Reply by ywmoyue** [reply]
确认一下是使用ccleaner清楚垃圾文件数据后必现？那我测一下

**Reply by Ech023** [reply]
https://github.com/ywmoyue/biliuwp-lite/assets/57533750/7ad90dde-4eff-4e03-aa45-a90645359ef4
你看看算是不

**Reply by Ech023** [reply]
@ywmoyue 我现在用的是最新的4.6.6.0版本

**Reply by ywmoyue** [reply]
那这样确实会清理掉应用给存的Cookie因为是调用的系统Cookie管理器的api来存的
这样的话用户自己用其他软件清理Cookie我也没法管，而且我看用户登录是已经没了的

**Reply by Ech023** [reply]
> 那这样确实会清理掉应用给存的Cookie因为是调用的系统Cookie管理器的api来存的
> 这样的话用户自己用其他软件清理Cookie我也没法管，而且我看用户登录是已经没了的
> 

之前橙子的版本清除好像不会删除登录信息这些，不知道是因为ccl是不是版本升级原因导致删除的

**Reply by ywmoyue** [reply]
@Ech023 麻烦验证一下4.6.9是否Cookie已不再会被CCleaner清除，需要重新登录一次

**Comment by zcg** [comment]
能否加个切换视频tab自动刷新评论的功能，现在切换视频tab不会刷新评论

**Reply by ywmoyue** [reply]
是个bug，下个版本恢复

**Comment by xiaoyanxyxy** [comment]
几千一万弹幕，显示不出效果了，无论怎么设置，都稀稀拉拉，只有三分之一，没有满屏的快感了

**Comment by Ech023** [comment]
https://github.com/ywmoyue/biliuwp-lite/assets/57533750/60203079-f8bf-4506-bb40-36d328881aae

 存在部分视频无法打开

**Reply by ywmoyue** [reply]
请单独创建一个帖子并将有问题的视频地址附上

**Comment by ybhgl** [comment]
最近几个版本直播为啥看1秒卡5秒，基本无法正常播放，网络没有问题，用逍遥橙子的聚合直播[https://github.com/xiaoyaocz/AllLive](url)就可以一直正常播放

**Reply by ywmoyue** [reply]
我自测了没问题，是从哪个版本开始的

**Reply by ybhgl** [reply]
具体记不清楚了，我从4.6.16一路往回装测试到4.6.7（卸载重装），都有这个问题，再早的版本似乎不能正常播放直播了
~感觉问题是缓冲区有问题？直播卡顿的时候网速不动，要等1-2s才会出现2mb/s左右的网速，然后再过1s左右才会加载出画面~
把直播链接抓出来直接播放也很卡，不知道是什么原因

**Reply by ybhgl** [reply]
另外发现一个小问题，弹幕引擎选择寒霜弹幕使，图里面的两个设置会消失，但是其实打开了加粗的选项再切回寒霜弹幕使，加粗的效果是会生效的，能否在选择寒霜弹幕使引擎时加入加粗的选项？
![20231004_102004](https://github.com/ywmoyue/biliuwp-lite/assets/35188529/a8e0824f-eef5-4c37-b54e-688532386330)

**Reply by ybhgl** [reply]
对比了一下本软件与聚合直播的直播链接，本软件是m3u8，聚合直播是flv

**Reply by ywmoyue** [reply]
@ybhgl 麻烦单独新开一个讨论贴把情况记录一下吧，在这个贴子里不方便管理记录

**Comment by GD-Slime** [comment]
是否可将直播的弹幕引擎也换为寒霜弹幕库？ 原有的NSdanmuku在弹幕量较高时会显示不全。

**Reply by ywmoyue** [reply]
确定是弹幕引擎影响的吗

**Reply by GD-Slime** [reply]
是的。 可以用一个简单的方法测试：直接打开某个弹幕测试视频，换用两个弹幕引擎，可以发现NS的明显少一些弹幕; 而寒霜已经弹幕多到堆叠。 直播弹幕也差不多是这样。

**Reply by GD-Slime** [reply]
或者，可以像视频弹幕一样提供弹幕引擎选择。

**Reply by ywmoyue** [reply]
@GD-Slime 那单独提个需求issue吧

**Reply by GD-Slime** [reply]
#341  开好了. 另外，是否也可以给寒霜弹幕库加上NSdanmuku才有的字体加粗和阴影样式呢？

**Reply by xiaoyaocz** [reply]
NSDanmaku不会堆叠，当屏幕没有空间时会被丢弃。我觉得堆叠的弹幕都看不清了，没啥意义

**Reply by GD-Slime** [reply]
> NSDanmaku不会堆叠，当屏幕没有空间时会被丢弃。我觉得堆叠的弹幕都看不清了，没啥意义

原来是这样设计的。我还以为是某种bug＞︿＜

**Reply by GD-Slime** [reply]
还有一个问题是，对于直播弹幕，不同于视频弹幕，是一波一波的更新的。 这导致每波弹幕并不能很好的占满整个屏幕，而总是会在后面留下一定的空隙; NSdanmuku每波弹幕如果多到堆叠，则被删除。这在大流量直播间下很容易发生。如果可以优化一下，将本列多出来的弹幕试图放到其后面的空隙里，直到空隙也装不下了再丢弃，或许可以很好的解决这个问题。

**Comment by xueshanqiu** [comment]
能不能出一个，只播放音频的功能，支持直播的那种

**Reply by GD-Slime** [reply]
这个的话是要听电台吗？ 那直接把直播开了挂后台也一样的

**Reply by xueshanqiu** [reply]
关键是占内存啊

**Comment by GD-Slime** [comment]
最近发现一直有这个问题：直播会不定时卡死。 
具体表现为：播放不定时后视频和声音均卡死，但此时弹幕仍然可以活动；直到点击页面上任意按钮后，整个app卡死，只能重启app.
卡死时日志没有什么异常。

**Comment by Genicur** [comment]
能不能加个评论区的关键字屏蔽，有的烂梗重复刷太多了，一点内容没有，纯纯的浪费精力。

**Comment by XKZKQY** [comment]
b站的稿件有的是包含多条高品质音轨的，如av113814922068396,有全景声和HIRES,如果缓存的话这个视频只能缓存一次，不能同时保存全景声和HIRES的音轨，可以开发同时保存全景声和HIRES的功能吗？或者可以让全景声和HIRES保存为两个视频吗？

**Reply by ywmoyue** [reply]
> b站的稿件有的是包含多条高品质音轨的，如av113814922068396,有全景声和HIRES,如果缓存的话这个视频只能缓存一次，不能同时保存全景声和HIRES的音轨，可以开发同时保存全景声和HIRES的功能吗？或者可以让全景声和HIRES保存为两个视频吗？

好注意，可以单独提个issue

**Reply by XKZKQY** [reply]
感谢！

**Reply by XKZKQY** [reply]
#1018

**Comment by heheheaa** [comment]
楼中楼的评论显示不全

**Comment by vs9ejfslvj32j4** [comment]
希望有可以直接复制视频标题的功能

**Reply by ywmoyue** [reply]
有的，右键视频标题可以复制

**Comment by christinaisme** [comment]
大佬，下载的时候在一个视频里选择了1080p，然后去下载另外一个视频，还是默认的360p，希望下载的画质可以延续之前一次的设定

**Comment by Fjjmyn** [comment]
大佬，能否提供屏蔽推荐页直播的选项吗？
![image](https://github.com/user-attachments/assets/5071d535-272f-4d1a-a047-9f9a44288df9)

**Reply by ywmoyue** [reply]
可以，单独提个issue吧

**Comment by Shaddoll126** [comment]
希望可以加入跳过赞助的功能，如b站空降助手

**Comment by Skyler1n** [comment]
希望能放大查看up主的头像

**Comment by lVentus** [comment]
快进或拖动进度条时经常无效，会跳回现在播放的时间点

**Reply by ywmoyue** [reply]
处理中

**Reply by lVentus** [reply]
> 处理中

新版本解决了，感谢

**Comment by sindw** [comment]
大佬,可以添加一个视频自动选择最高画质的功能吗

**Reply by ywmoyue** [reply]
你可以在一个8K的视频下选择8K分辨率画质，后面不再动这个设置的话就是自动选择最高分辨率（HDR和杜比另外算

**Comment by sindw** [comment]
大佬好,下载功能有个问题,如果客户端长时间窗口没有在最前面,下载的视频会被冻结(就算没有冻结下载速度也会明显下降),一般会提示因系统原因暂停,也有可能是直接提示冻结.而且如果是系统暂停的话,直接关掉客户端再打开,下载就必须重新开始,不然直接开始之前被暂停的下载,客户端会闪退

**Reply by ywmoyue** [reply]
下载用的系统本身的后台下载服务，应该不会因为应用不在前端就冻结，有个问题是如果下载时间超过两小时那么播放源地址是会失效的，你看看下载时间用了多久

**Reply by sindw** [reply]
> 下载用的系统本身的后台下载服务，应该不会因为应用不在前端就冻结，有个问题是如果下载时间超过两小时那么播放源地址是会失效的，你看看下载时间用了多久

感谢大佬的回复,下载时间如果我放在前台的话一直下都没问题(最长的一次下载了一个半小时没有事),后面以防链接有问题我还清空下载,重新申请了下载请求,仅仅缩到后台十分钟不到就冻结了

**Reply by sindw** [reply]
> 下载用的系统本身的后台下载服务，应该不会因为应用不在前端就冻结，有个问题是如果下载时间超过两小时那么播放源地址是会失效的，你看看下载时间用了多久

不好意思啊大佬,我找到问题了,是我的原因.我在设置里面启用了"使用下载索引"这个功能,关闭这个功能之后就正常了

**Reply by ywmoyue** [reply]
那说明启用这个功能后还是有bug

**Comment by System32laoma** [comment]
> 如果是代码上的问题，请提交issue 本贴是公告，尽量不要在本贴提bug

想在设置里增添一个选项，能让评论区的网页链接显示整个网址，而不是“网页链接”这几个字

**Reply by ywmoyue** [reply]
鼠标悬浮上去可以显示整个网址

**Reply by System32laoma** [reply]
> 鼠标悬浮上去可以显示整个网址

就是吧，有的人在评论区发代码，而代码里有一行是网址，这样的话就有些别扭
<img width="622" height="346" alt="image" src="https://github.com/user-attachments/assets/03fe8066-1ecb-48b5-a01e-d3ed0a0c09ce" />

**Reply by ywmoyue** [reply]
哦牛的，还有这种

**Reply by System32laoma** [reply]
> 哦牛的，还有这种

🤭

**Comment by System32laoma** [comment]
> 如果是代码上的问题，请提交issue 本贴是公告，尽量不要在本贴提bug

空降助手可以增添手动跳过的按钮吗，就是到可以空降的进度后出现一个提示按钮，点击按钮后执行空降（参考biliplus）

**Reply by ywmoyue** [reply]
有空看下，我记得目前有策略就是非广告的标记片段就是你说的这种

**Reply by GD-Slime** [reply]
是的, 目前标记为非广告片段的其他片段就会显示跳过按钮.
在标记是否有跳过按钮的地方加个选项就行了

**Reply by System32laoma** [reply]
> 有空看下，我记得目前有策略就是非广告的标记片段就是你说的这种

行，因为都的时候有人标记的不是特别枯燥无味的广告我还是想看看，弄个按钮可以让。用户自己决定

**Comment by System32laoma** [comment]
> 如果是代码上的问题，请提交issue 本贴是公告，尽量不要在本贴提bug

大佬，B站对评论的风控太严重了，有的时候自己的评论被shadow ban（仅自己可见）了都不知道，大佬能否加一个评论反诈功能，这个也是可以参考biliplus

**Reply by ywmoyue** [reply]
这个功能有其他开源项目的相关issue或提交记录吗

**Reply by System32laoma** [reply]
![Screenshot_2025-09-11-14-28-00-750_com yujian ResideMenuDemo](https://github.com/user-attachments/assets/db09b263-0f81-440e-b0ea-0600efcdbe70)
[Uploading PiliPlus-1.1.3.15.zip…]()
[Uploading PiliPlus-1.1.3.6.zip…]()

这里有，但是哔哩发评反诈是安卓端，这里也有关于哔哩发评反诈的链接，有源码，然后我看1.1.3.6应该是变成软件内置了（这样应该工程量小点）

**Reply by System32laoma** [reply]
> 这个功能有其他开源项目的相关issue或提交记录吗

这两个链接在这里，有1.1.3.15和1.1.3.6
https://github.com/bggRGjQaUbCoE/PiliPlus/releases?q=%E5%8F%8D%E8%AF%88&expanded=true

**Reply by ywmoyue** [reply]
好的，我看看

**Reply by System32laoma** [reply]
> 好的，我看看

行，他这个界面是这样的（还有别的，比如正常评论就会说无账号检测到评论，正常显示）
![Screenshot_2025-09-11-15-08-49-408_com example piliplus](https://github.com/user-attachments/assets/6b16b7c6-0587-402b-9229-fadff76d1722)

**Comment by wwhdgithub** [comment]
版本4.8.6，每个视频都会自动打开CC字幕，在设置页关了自动打开AI字幕也还是这样

**Reply by ywmoyue** [reply]
更新4.8.6.702

**Comment by mingqiannailv** [comment]
不知道为什么空降助手是一直失效的，在网页端/其他有空降助手功能的第三方端都能跳过，但是这边永远是没有可跳过片段

**Reply by ywmoyue** [reply]
我这边正常，方便的话可以发下日志我看看

**Reply by mingqiannailv** [reply]
<img width="1891" height="936" alt="8bf0d9af-13c1-4175-b52a-db215c7c99d3" src="https://github.com/user-attachments/assets/bfa10b1b-1d1b-45d5-9c0b-9aafe077e2ad" />
<img width="1625" height="665" alt="33c45f52-2f97-4d1c-abf8-fa65bf2dbf1a" src="https://github.com/user-attachments/assets/aeee10fc-a3dd-45c7-9b89-de1ba9710cc3" />
[20260212.log](https://github.com/user-attachments/files/25266986/20260212.log)

非常奇怪，这个示例视频都会提示无片段，但是网页端就很正常
<img width="1003" height="674" alt="89c5a432-21c1-449b-963d-1a5a6d94f3b2" src="https://github.com/user-attachments/assets/632e7f8b-01f4-418f-9c25-1536f621c72a" />

**Reply by ywmoyue** [reply]
你的日志没传上来，我访问是404的，不过我刚才试了十几次其中复现了一次，检查了下代码这块确实可能因为非同步的流程导致控件中拿不到数据，我晚点处理下
